File: test-npm-pkg/ts/tests/test-agent.ts
import { pathToFileURL } from 'url';
import { client, runTestSection, logResult } from "../utils/utils.js";

async function testAgent() {
  console.log("\n🚀 Testing Agent Templates and Use Cases...");

  let templateId: string | null = null;
  let useCaseId: string | null = null;

  // 1. Templates
  await runTestSection("agent.list", async () => {
    const templates = await client.agent.list();
    logResult("Templates Count", templates.length);
  });

  await runTestSection("agent.create (Custom)", async () => {
    const template = await client.agent.create({
      assistant: {
        name: `SDK Test Agent ${Date.now()}`,
        instructions: "You are a test agent."
      },
      usecase: {
        name: `SDK Test UseCase ${Date.now()}`,
        category: "Test"
      }
    });
    templateId = template._id || template.id;
    logResult("Template Created", templateId);
  });

  if (templateId) {
    await runTestSection("agent.get", () => client.agent.get(templateId!));
    await runTestSection("agent.update", () => client.agent.update(templateId!, {
        assistant: { instructions: "Updated instructions" }
    }));
    await runTestSection("agent.delete", () => client.agent.delete(templateId!));
  }

  // 2. Use Cases
  await runTestSection("agent.listUseCases", async () => {
    const useCases = await client.agent.listUseCases();
    logResult("UseCases Count", useCases.length);
  });

  await runTestSection("agent.createUseCase", async () => {
    const uc = await client.agent.createUseCase({
      name: `SDK Custom UC ${Date.now()}`,
      description: "Test description"
    });
    useCaseId = uc._id || uc.id;
    logResult("UseCase Created", useCaseId);
  });

  if (useCaseId) {
    await runTestSection("agent.getUseCase", () => client.agent.getUseCase(useCaseId!));
    await runTestSection("agent.updateUseCase", () => client.agent.updateUseCase(useCaseId!, {
        description: "Updated UC description"
    }));
    await runTestSection("agent.deleteUseCase", () => client.agent.deleteUseCase(useCaseId!));
  }

  console.log("\n✅ Agent Resource Testing Completed.");
}

if (import.meta.url === pathToFileURL(process.argv[1]).href) {
  testAgent().catch(console.error);
}

export { testAgent };

File: test-npm-pip-pkg/py/tests/test_agent.py
import time
from utils.utils import client, run_test_section, log_result

def test_agent():
    print("\n🚀 Testing Agent Templates and Use Cases...")

    state = {"template_id": None, "use_case_id": None}

    # 1. Templates
    def templates_ops():
        templates = client.agent.list()
        log_result("Templates Count", len(templates))

        created = client.agent.create({
            "assistant": {
                "name": f"SDK Test Agent {int(time.time())}",
                "instructions": "You are a test agent.",
                "model": "gpt-4o-mini"
            },
            "usecase": {
                "title": f"SDK Test UseCase {int(time.time())}",
                "category": "Test"
            }
        })
        state["template_id"] = created.get("_id") or created.get("id")
        log_result("Template Created", state["template_id"])

        if state["template_id"]:
            client.agent.get(state["template_id"])
            client.agent.update(state["template_id"], {
                "assistant": {"instructions": "Updated instructions"}
            })
            log_result("Template Ops Verified", True)

            client.agent.delete(state["template_id"])
            log_result("Template Deleted", True)
    run_test_section("Templates Ops", templates_ops)

    # 2. Use Cases
    def use_cases_ops():
        use_cases = client.agent.list_use_cases()
        log_result("UseCases Count", len(use_cases))

        created = client.agent.create_use_case({
            "name": f"SDK Custom UC {int(time.time())}",
            "description": "Test description"
        })
        state["use_case_id"] = created.get("_id") or created.get("id")
        log_result("UseCase Created", state["use_case_id"])

        if state["use_case_id"]:
            client.agent.get_use_case(state["use_case_id"])
            client.agent.update_use_case(state["use_case_id"], {
                "description": "Updated UC description"
            })
            log_result("UseCase Ops Verified", True)

            client.agent.delete_use_case(state["use_case_id"])
            log_result("UseCase Deleted", True)
    run_test_section("Use Cases Ops", use_cases_ops)

    print("\n✅ Agent Resource Testing Completed.")

if __name__ == "__main__":
    test_agent()
